File: Maven/KotlinTutorial/src/main/kotlin/com/example/KotlinTutorial/anonymousInnerClass.java
package com.example.KotlinTutorial;

//  anonymous inner class
interface Human2{
    void think();
}

public class anonymousInnerClass {
    public static void main (String[] args){

        Human2 programmer = new Human2() {      //  here "new" will not work bcoz, Human2 is an interface
//            to make it work we can directly make an anonymous inner class
            public void think() {
                System.out.println("Think virtually...");
            }
        };

        programmer.think();
    }
}

File: Maven/KotlinTutorial/src/main/kotlin/com/example/KotlinTutorial/anonymousInnerClass.kt
interface Human2{
    fun think()
}

fun main(args: Array<String>){
//    var programmer: Human2 = Human2()   // this can't be done, bcoz this needs a class name
    var programmer: Human2 = object : Human2 // this is the syntax for anonymous inner class in Kotlin
//                          using the object keyword we're creating the instance only once and this instance is not reusable
                            {
                                override fun think() {
                                    println("Think virtually...")
                                }
                            }
    programmer.think()
}